tests/e2e/scenarios/test_provider_capability_inventory.py

```
@@ -396,6 +535,25 @@ def test_journey_evidence_rejects_a_vanished_symbol(
+def test_journey_evidence_rejects_a_same_named_local_stub():
+    source = (
+        "from provider_journey_google import assert_google_provider_outcome\n"
+        "\n"
+        "async def assert_google_provider_outcome(url, calls):\n"
+        "    return None\n"
+        "\n"
+        "async def test_journey(url, calls):\n"
+        "    await assert_google_provider_outcome(url, calls)\n"
+    )
+
+    with pytest.raises(AssertionError, match="binding .* is shadowed"):
```

Ruff (0.16.0)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)